sdks/java/io/cdap/src/main/java/org/apache/beam/sdk/io/cdap/Plugin.java

```diff
@@ -84,16 +88,34 @@ public PluginConfig getPluginConfig() {
    * validating connection to the CDAP sink/source and performing initial tuning.
    */
   public void prepareRun() {
+    PluginConfig pluginConfig = getPluginConfig();
+    if (pluginConfig == null) {
+      throw new IllegalArgumentException("PluginConfig should be not null!");
+    }
     if (cdapPluginObj == null) {
-      InstantiatorFactory instantiatorFactory = new InstantiatorFactory(false);
-      cdapPluginObj =
-          (SubmitterLifecycle) instantiatorFactory.get(TypeToken.of(getPluginClass())).create();
+      for (Constructor<?> constructor : getPluginClass().getDeclaredConstructors()) {
```

Comment:
this is odd to me. It seems like we shouldn't be grabbing all the constructors, but just the one that can be constructed with the plugin config.

If we have multiple constructors here, one will certainly cause the exception block to be hit, even if the other constructor is valid

Reply:
You're right, there is no need to grab all the constructors here.
Fixed.